lib/tasks/asset_sync.rake

```diff
@@ -10,21 +9,4 @@ namespace :assets do
       AssetSync.clean
     end
   end
-
-end
-
-if Rake::Task.task_defined?("assets:precompile:nondigest")
-  Rake::Task["assets:precompile:nondigest"].enhance do
-    # Conditional execution needs to be inside the enhance block because the enhance block
-    # will get executed before yaml or Rails initializers.
-    Rake::Task["assets:sync"].invoke if defined?(AssetSync) && AssetSync.config.run_on_precompile
-  end
-else
-  Rake::Task["assets:precompile"].enhance do
-    # rails 3.1.1 will clear out Rails.application.config if the env vars
-    # RAILS_GROUP and RAILS_ENV are not defined. We need to reload the
-    # assets environment in this case.
-    # Rake::Task["assets:environment"].invoke if Rake::Task.task_defined?("assets:environment")
-    Rake::Task["assets:sync"].invoke if defined?(AssetSync) && AssetSync.config.run_on_precompile
```

Comment:
are you running `assets:sync` separately in your project? I needed to make a change so that this task is run after `webpacker:compile` (when defined). See #341 for my solution, I will update it so it works with Webpacker 2.0 - inspired by your changes